### PR TITLE
This removes the second CacheSM, and related APIs

### PR DESCRIPTION
**Pull request by zwoop**
For some details as to why this is important, see:

    https://cwiki.apache.org/confluence/display/TS/Cleanup+of+Cache+URLs

Note that this leaves the stale_while_revalidate plugin in a broken state,
but I rather keep the code there for now, since there's hope someone has a
fix for this plugin, which does not require this now removed API.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Sent an email to the mailing list for discussion / approval as well.

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -2438,19 +2434,15 @@ HttpSM::state_cache_open_write(int event, void *data)
     ink_release_assert(0);
   }
 
+  // ToDo: Can this, at this point, be anything but LOCK_URL_FIRST ?
```
Comment (SolidWallOfCode) [inline proxy/http/HttpSM.cc]: Perhaps put an assert in the `else` to verify that doesn't happen?

**Comment by zwoop** [comment]
I eliminated the rest of the "state" tracking, since LOCK_URL_ORIGINAL was eliminated long ago, there really is only one state left here, the "FIRST URL".

**Comment by zwoop** [comment]
I've been running with this on Docs / CI for a day now, so far so good.

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**